src/main/java/edu/ucsb/cs56/mapache_search/controllers/SearchController.java

```diff
@@ -142,6 +148,32 @@ public String search(SearchParameters params, Model model, OAuth2AuthenticationT
         Long time = userRepository.findByUid(controllerAdvice.getUid(token)).get(0).getTime();
         Long currentTime = (long) (new Date().getTime()/1000/60/60/24); //get relative days as a Long
 
+        boolean haveSearched = isSearchExist(params.getQuery());
+        if (!haveSearched) // Have never been searched before
+        {
+            SearchTerms searchTerm = new SearchTerms();
+            searchTerm.setSearchTerms(params.getQuery());
+            searchTerm.setCount(1);
+            searchTerm.setTimestamp(new Date());
+            searchTermsRepository.save(searchTerm);
+            System.out.println("count is: " + searchTerm.getCount());
```

Comment:
Let's make these use the logger instead of system.out.println. there's some examples of this throughout the code, just search for "logger". also, let's put it all on one line